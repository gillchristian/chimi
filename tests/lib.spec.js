const { stripIndent } = require('common-tags')
const { injectDependencies } = require('../lib/lib')

describe('lib', () => {
  describe('injectDependencies', () => {
    const code = stripIndent`
      const answer = 42

      console.log(answer)
    `

    it('should not change the input when no dependencies are passed', () => {
      const dependencies = {}
      const result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()
    })

    it('should prepend requires with variable declarations to the input', () => {
      let dependencies = {
        lodash: '_',
      }
      let result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()

      dependencies = {
        lodash: '_',
        trae: 'trae',
      }
      result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()
    })

    it('should work for local dependencies', () => {
      const dependencies = {
        './lib': 'lib',
        '../foo/bar': 'bar',
      }
      const result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()
    })

    it('should not add a variable declaration when values are falsy', () => {
      const dependencies = {
        './for-the-side-effects': false,
        'es6-promise': false,
      }
      const result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()
    })
  })
})
