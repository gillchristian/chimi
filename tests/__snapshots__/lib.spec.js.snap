// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib injectDependencies local dependency 1`] = `
"// snippet dependencies
let lib = require('./lib');

const answer = 42

console.log(answer)"
`;

exports[`lib injectDependencies no dependencies 1`] = `
"// snippet dependencies


const answer = 42

console.log(answer)"
`;

exports[`lib injectDependencies one dependency 1`] = `
"// snippet dependencies
let _ = require('lodash');

const answer = 42

console.log(answer)"
`;

exports[`lib injectDependencies two dependencies 1`] = `
"// snippet dependencies
let _ = require('lodash');
let trae = require('trae');

const answer = 42

console.log(answer)"
`;
